packit/git_utils.py

```diff
@@ -109,3 +92,37 @@ def get_metadata_from_message(commit: git.Commit) -> Optional[dict]:
             return loaded_part
 
     return None
+
+
+def git_patch_ish(patch: str) -> str:
+    """
+    Massage patch to look like a Git-style patch, so that it can
+    be passed to 'git patch-id' in order to calculate a patch-id.
+    """
+    # Prettend as if format is 'diff --git'
+    pattern = re.compile(r"^diff -\w+ ", flags=re.MULTILINE)
+    repl = r"diff --git "
+    patch = re.sub(pattern, repl, patch)
+
+    # Remove timestamps from comparison lines
+    pattern = re.compile(r"^((---|\+\+\+) .+)\t\d{4}.+$", flags=re.MULTILINE)
+    repl = r"\1"
+    patch = re.sub(pattern, repl, patch)
+
+    # Add missing 'diff --git' lines
```

Comment:
Can the 2. and 3. blocks of code be swapped?
I'd spent less brainpower for switching from 'diff --git' to timestamps and back to 'diff --git'

Reply:
With a changed order the regexp pattern for the substitution to add missing diff lines would become more complicated, I consider reading regexp requiring more brainpower than a strange ordering of blocks :) Added a note that the order matters.